app/Models/UserReport.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ *    Copyright 2015-2018 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Models;
+
+use App\Models\Score\Best\Model as BestModel;
+
+class UserReport extends Model
+{
+    const CREATED_AT = 'timestamp';
+
+    protected $table = 'osu_user_reports';
+    protected $primaryKey = 'report_id';
+    protected $guarded = [];
+
+    protected $dates = ['timestamp'];
+
+    public $timestamps = false;
+
+    public function reporter()
+    {
+        return $this->belongsTo(User::class, 'reporter_id');
+    }
+
+    public function score()
+    {
+        return $this->belongsTo(BestModel::getClass($this->mode), 'score_id');
```

Comment:
as I recently learned, using `$this` in association definition doesn't work for `with()` preloads.

Reply:
could use `morphTo` except the `int` mode breaks things especially when it's `0`

Reply:
I don't think it'll be an issue, yet; to use `morphTo` we'll probably have to change the data type, otherwise we might have to make a separate association for each `mode` or something :|